wolfcrypt/src/evp.c

```diff
@@ -3291,15 +3327,57 @@ int wolfSSL_EVP_PKEY_sign(WOLFSSL_EVP_PKEY_CTX *ctx, unsigned char *sig,
             *siglen = (size_t)len;
             return WOLFSSL_SUCCESS;
         }
-        /* wolfSSL_RSA_sign_generic_padding performs a check that the output
-         * sig buffer is large enough */
-        if (wolfSSL_RSA_sign_generic_padding(wolfSSL_EVP_MD_type(ctx->md), tbs,
-                (unsigned int)tbslen, sig, &usiglen, ctx->pkey->rsa, 1,
-                ctx->padding) != WOLFSSL_SUCCESS) {
-            return WOLFSSL_FAILURE;
+
+#if defined(OPENSSL_ALL) && (!defined(HAVE_FIPS) || FIPS_VERSION_GT(2,0))
+        /* Handle PSS padding using RSA_padding_add_PKCS1_PSS_mgf1 if saltlen
+         * or mgf1 hash were set.  Use generic signing otherwise. */
+        if (ctx->mgf1_md || ctx->saltlen) {
+            int ret;
+            unsigned char *encodedSig = NULL;
+            int emLen = wolfSSL_RSA_size(ctx->pkey->rsa);
+            int saltLen = ctx->saltlen;
+            const WOLFSSL_EVP_MD *mgf1Hash = ctx->mgf1_md;
+
+            /* Allocate memory for encoded signature */
+            encodedSig = (unsigned char *)XMALLOC(emLen, NULL,
+                                                 DYNAMIC_TYPE_TMP_BUFFER);
+            if (encodedSig == NULL)
+                return WOLFSSL_FAILURE;
+
+            /* Create PSS padded signature */
+            ret = wolfSSL_RSA_padding_add_PKCS1_PSS_mgf1(ctx->pkey->rsa,
```

Comment:
Odd this function doesn't use WOLFSSL_SUCCESS or WOLFSSL_FAILURE... Consider fixing and using it